address/records.go

```diff
@@ -41,7 +41,7 @@ const (
 
 func newAddressVersionRecord(version *asset.Version) tlv.Record {
 	return tlv.MakeStaticRecord(
-		addrVersionType, version, 1, asset.VersionEncoder,
+		addrAssetVersionType, version, 1, asset.VersionEncoder,
```

Comment:
Open question: is there a case where we'd want to bump the version of the asset (to be created), but keep the addr version the same? I think you've been hinting at some of these questions re the over proliferation of versions: do we actually need them all, and for the places we already have them, do we have a set plan for usage?

Reply:
Related question is if we have just the addr version: what is the primary use of such a version? Let's say we were to do some of the compression stuff, but having the internal key and the script key be the same. Do we need a new version there? Or is the implicit knowledge that if only one is specified, it applies for both fields enough?

Reply:
> implicit knowledge that if only one is specified, it applies for both fields enough?

I think that implicit knowledge makes documentation complicated. A version field is simpler.

I can think of certain addr specific updates that we might want to implement in the future that shouldn't involve a change in asset version. Where the simplest path would involve bumping a Tap address version. Adding a secondary proof courier address (after Tap address compression) is one possible feature that we've discussed briefly in the past: https://github.com/lightninglabs/taproot-assets/issues/461 .

We might also want to include a new expiration field within a Tap address for example. So that a receiver has the option of issuing a new refreshed address at a later date. This gives the receiver the option to stop using a proof courier that they've previously encoded into an address. Which might come in handy if a receiver has publicly shared a repeat use address (where sending nodes use a local saved copy of the address), but the address specified proof courier service is suddenly refusing to cooperate.

Reply:
Agree, and right now we also don't have a flexible field in the Address like the Metadata field we have for Assets, so the room to add in other fields in the future is limited without some versioning.